---
title: "Analysis of PacMAN metabarcoding data"
author: "Saara Suominen, Pieter Provoost"
output: 
  html_document:
    theme: flatly
    toc: yes
    toc_float:
      collapsed: false
editor_options: 
  chunk_output_type: console
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, encoding = encoding, output_dir = "../docs") })  
---

This is a first analysis of the COI metabarcording data from PacMAN available as of July 2023. Raw sequence data have been analyzed with the [PacMAN bioinformatics pipeline](https://github.com/iobis/PacMAN-pipeline) and the resulting Darwin Core tables are included in this repository.

The source for this notebook is hosted at <https://github.com/iobis/pacman-data-analysis>.

![](https://github.com/iobis/pacman-data-analysis/actions/workflows/deploy.yml/badge.svg)

```{r setup, include=FALSE, cache=FALSE}
knitr::opts_knit$set(
  root.dir = rprojroot::find_rstudio_root_file()
)
```

```{r warning=FALSE, message=FALSE, echo=FALSE}
library(dplyr)
library(ggplot2)
library(leaflet)
library(sf)
library(rmarkdown)
library(stringr)
library(readr)
library(viridis)
```

```{r echo=FALSE}
occurrence <- read.table("data/COI/Occurence_table.tsv", sep = "\t", header = TRUE, na.strings = "") %>%
  mutate(
    species = ifelse(taxonRank == "species", scientificName, NA),
    eventDate = parse_date(str_match(eventID, "_([0-9]+)_")[,2], format = "%Y%m%d"),
    eventType = case_when(
      str_detect(eventID, "_P") ~ "plankton",
      str_detect(eventID, "_S") ~ "plate",
      str_detect(eventID, "_W") ~ "water"
    )
  )

dna <- read.table("data/COI/DNA_extension_table.tsv", sep = "\t", header = TRUE, na.strings = "")
```

## Sample overview

```{r warning=FALSE, message=FALSE, echo=FALSE}
samples <- occurrence %>%
  filter(!is.na(decimalLongitude) & !is.na(decimalLatitude)) %>%
  group_by(locationID, eventID, materialSampleID, eventType, decimalLongitude, decimalLatitude, eventDate) %>%
  summarize(asvs = n(), reads = sum(organismQuantity)) %>%
  arrange(locationID, eventDate, eventType, materialSampleID) %>%
  st_as_sf(coords = c("decimalLongitude", "decimalLatitude"), crs = 4326)

leaflet(samples, width = "100%") %>% addTiles() %>% addMarkers(
  clusterOptions = markerClusterOptions(maxClusterRadius = 10),
  popup = ~eventID
)
```

```{r warning=FALSE, message=FALSE, echo=FALSE, fig.width=12}
ggplot(samples) +
  geom_bar(aes(x = locationID, fill = eventType), width = 0.3) +
  scale_fill_manual(values = c(plankton = "#8BBEB2", water = "#456092", plate = "#18314F")) +
  ggtitle("Number of samples by location and sample type")
```

```{r warning=FALSE, message=FALSE, echo=FALSE}
samples %>%
  st_drop_geometry() %>%
  paged_table(options = list(rows.print = 20))
```

## Taxonomic composition

```{r warning=FALSE, message=FALSE, echo=FALSE, fig.height=15, fig.width=12}
stats_phylum <- occurrence %>%
  filter(locationID != "Control" & !is.na(phylum)) %>%
  group_by(phylum, locationID, eventType) %>%
  summarize(reads = sum(organismQuantity)) %>%
  ungroup()

ggplot(data = stats_phylum) +
  geom_bar(aes(y = phylum, x = reads, fill = phylum), stat = "identity") +
  scale_x_continuous(trans = "log10") +
  scale_fill_viridis(discrete = TRUE, option = "magma", begin = 0.2, end = 0.8) +
  scale_y_discrete(limits = rev) +
  facet_grid(eventType~locationID) +
  theme(legend.position = "none") +
  ggtitle("Number of reads by location, sample type, and phylum")
```

## Introduced species

To do.

## Full species list

```{r warning=FALSE, message=FALSE, echo=FALSE}
species <- occurrence %>%
  filter(!is.na(species)) %>%
  group_by(phylum, class, species) %>%
  summarize(
    reads = sum(organismQuantity),
    water = "water" %in% eventType,
    plate = "plate" %in% eventType,
    plankton = "plankton" %in% eventType
  ) %>%
  arrange(desc(reads))

species %>%
  paged_table(options = list(rows.print = 20))
```
